src/bin/lockin.rs

```diff
@@ -137,29 +157,57 @@ const APP: () = {
             c.resources.adcs.1.acquire_buffer(),
         ];
 
-        let dac_samples = [
+        let mut dac_samples = [
             c.resources.dacs.0.acquire_buffer(),
             c.resources.dacs.1.acquire_buffer(),
         ];
 
         let lockin = c.resources.lockin;
         let settings = c.resources.settings;
 
-        let timestamp =
-            c.resources.timestamper.latest_timestamp().unwrap_or(None); // Ignore data from timer capture overflows.
-        let (pll_phase, pll_frequency) = c.resources.pll.update(
-            timestamp.map(|t| t as i32),
-            settings.pll_tc[0],
-            settings.pll_tc[1],
-        );
+        let mut pll_frequency = 0;
+
+        let (sample_phase, sample_frequency) = match settings.lockin_mode {
+            LockinMode::External => {
+                let timestamp =
+                    c.resources.timestamper.latest_timestamp().unwrap_or(None); // Ignore data from timer capture overflows.
+                let (pll_phase, frequency) = c.resources.pll.update(
```

Comment:
Probably better to call these `reference_phase` and `reference_frequency` and let both arms return `reference_frequency` (constant in the `Internal` case) that's then used as `conf::ReferenceFrequency` output.